W-15535654 Updates to improve SSO content jh (#32)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/sso-use-case-1.adoc

= Configuring SSO When No Identity Provider Is Configured in Anypoint Platform
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> This use case includes instructions for setting up an identity provider for your portal when no other identity provider is configured in Anypoint Platform. You want to configure an identity provider as a login option for your portal. Since there is no identity provider configured in Anypoint Platform, the default identity provider automatically maps the identity provider for the portal to the identity provider in Anypoint Platform. This enables every Salesforce user to have an identity in Anypoint Platform. See xref:enabling-sso-for-your-portal.adoc#out-of-the-box-configurations[Out-of-the-Box Configurations] for more information about the default identity provider.

Reviewer: ```suggestion
If you haven't configured another identity provider in Anypoint Platform, you can use the default identity provider. The default provider automatically maps the identity provider for the portal to the identity provider in Anypoint Platform, providing every Salesforce user an identity in Anypoint Platform. See xref:enabling-sso-for-your-portal.adoc#out-of-the-box-configurations[Out-of-the-Box Configurations] for more information about the default identity provider.
```

Author: Hi @hannanelson, I think "...you can use the default identity provider" may imply that you can choose to use it. This is not the case here. This happens automatically if aeh_portal_idp is not set in the identity provider.

== Before You Begin

> * In the identity provider of your choice, the users or identities must be created for the group of the users who have access to the application. 

Reviewer: ```suggestion
* In the identity provider of your choice, the users or identities must exist for the users with access to the application. 
```

> * Review the information in the xref:enabling-sso-for-your-portal.adoc#before-you-begin [Before You Begin] section to understand the roles and permissions required and the information that API Experience Hub expects to configure custom user attributes and claims for the identity provider.

Reviewer: ```suggestion
* Review the information in the xref:enabling-sso-for-your-portal.adoc#before-you-begin [Before You Begin] section to understand the roles, permissions, and other information that API Experience Hub requires to configure custom user attributes and claims for the identity provider.
```

Reviewer: In this sentence, expects is unclear to me. I read it as AEH expects (anticipates) to configure, not "stuff AEH expects (requires) to be able to configure".

. In Okta, create an OpenID Connect Web application. For more information, see https://help.okta.com/oie/en-us/content/topics/apps/apps_app_integration_wizard_oidc.htm[Create OIDC app integrations^].
. From the *General Settings* section, complete the following fields:
+
[%header,cols="1a,2a"]
|===
|Field|Value
|*App integration name*|Enter a name for the app.
|*Grant type* |Select *Authorization Code*.
|*Assignments* |Select *Limit access to selected groups*.
|*Selected group(s)*|Enter the name of the group who must have access to the application.
|===

> . Verify that your identity provider sends the expected claims listed in the xref:enabling-sso-for-your-portal.adoc#before-you-begin[Before You Begin] section. 

Reviewer: Nice!

. Configure the claims that the application sends.
+

> For example, for OpenID Connect for Okta, you can configure the groups claim using the following steps:

Reviewer: ```suggestion
For example, for OpenID Connect for Okta, configure the groups claim using the following steps:
```

Author: groups claim

|*Selected group(s)*|Enter the name of the group who must have access to the application.
|===

> . Configure a custom groups claim filter that the application sends.
> +
> If you are not using OpenID Connect, review the xref:enabling-sso-for-your-portal.adoc#before-you-begin[Before You Begin] section for the values that API Experience expects for claims and attributes and the documentation for the identity provider. 
> 

Reviewer 2: ```suggestion
. Configure required claims that the application sends.
+
Review the xref:enabling-sso-for-your-portal.adoc#before-you-begin[Before You Begin] section to verify that your Identity Provider send the expected claims.
For example, for OpenID in Okta you can configure the groups claim by doing the following:
```

Author: @nbattiato Should you review the Before You Begin section before the step in line 41 Configure required claims that the application sends?

Reviewer 2: @JennyHajee my point with this comment is to transform this step a where we only talk about the **groups** claim to step where the user needs to configure all the claims, pointing to the before you begin section and giving the example of the groups claim. All other required claims are skipped since the default claims Okta sends are the included in our defaults

Author: Thanks @nbattiato, I think it makes sense to move the step to review the Before you begin section before you configure the claims.

.. In *Salesforce*, go to *Setup*.
.. In the *Quick Find* box, enter `Auth`, and then select *Auth. Providers*.
.. Click *New*.

> .. In the *Provider Type* field, select *OpenID Connect*.

Reviewer: Is this a menu? Maybe just remove say "For *Provider Type*, select"?

Author: done

|===
+
image::aeh-sso-saml-settings.png["SAML Settings example",width=90%]

> . Review the xref:enabling-ss0-for-your-portal#before-you-begin[Before You Begin] section for information about what API Experience Hub expects for attributes to properly map users in the identity provider.

Reviewer 2: ```suggestion
. Review the xref:enabling-ss0-for-your-portal#before-you-begin[Before You Begin] section for information about what API Experience Hub expects for claims to properly map users in the identity provider.
```
